homeassistant/components/filesize/sensor.py

```diff
@@ -12,69 +12,71 @@
     PLATFORM_SCHEMA as PARENT_PLATFORM_SCHEMA,
     SensorEntity,
 )
-from homeassistant.const import DATA_MEGABYTES
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_FILE_PATH, DATA_MEGABYTES
 from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.reload import setup_reload_service
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import DOMAIN, PLATFORMS
+from .const import CONF_FILE_PATHS, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-
-CONF_FILE_PATHS = "file_paths"
 ICON = "mdi:file"
 
 PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
     {vol.Required(CONF_FILE_PATHS): vol.All(cv.ensure_list, [cv.isfile])}
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the file size sensor."""
-
-    setup_reload_service(hass, DOMAIN, PLATFORMS)
-
-    sensors = []
-    paths = set()
+    _LOGGER.warning(
+        # Filesize config flow added in 2022.4 and should be removed in 2022.8
+        "Loading filesize via platform setup is deprecated; Please remove it from your configuration"
```

Comment:
This can be a bit more extensive and made clear for the end-user.

For example:

```py
        "Configuration of the Filesize sensor platform in YAML is deprecated and "
        "will be removed in Home Assistant 2022.6; Your existing configuration "
        "has been imported into the UI automatically and can be safely removed "
        "from your configuration.yaml file"
```

This add clarity of what the user can do, and also is re-assuring it has been automatically handled/done.